OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs

```diff
@@ -118,7 +118,7 @@ public void UpdateMouseover()
 			}
 
 			var frozen = world.ScreenMap.FrozenActorsAt(world.RenderPlayer, worldRenderer.Viewport.ViewToWorldPx(Viewport.LastMousePos))
-				.Where(a => a.TooltipInfo != null && a.IsValid)
+				.Where(a => a.TooltipInfo != null && a.IsValid && !a.Actor.TraitsImplementing<IVisibilityModifier>().Any(t => !t.IsVisible(a.Actor, world.RenderPlayer)))
```

Comment:
Maybe there should be a method for this check (maybe an extension method), which this and ```CanViewActor``` could both use.